### PR TITLE
Email scheduling

### PR DESCRIPTION
**Pull request by jgfino**
### ℹ️ Issue

Closes [clickup ticket link](https://app.clickup.com/t/25czuf6)

### 📝 Description

- Added a ``sent`` column to the ``assignments`` table in the database to keep track of which assignments have already been sent to youth
- Added email scheduling using ``@nestjs/schedule`` so that letter PDFs are sent to treatment youth for completed assignments every day at 5:00pm.
- Had to rewrite ``amazon-ses.wrapper`` to allow sending email attachments. It now uses the ``sendRawEmail`` function and ``nodemailer``'s email generator. Functionality and arguments are the same, but now there is another argument ``attachments`` for an array of base64 encoded attachments to be sent with an email

### ✔️ Verification

- Wrote a unit test for the letter sending function to make sure it only sends completed, unsent assignments to only treatment youth
- Manually verified email and attachment
- Manually verified scheduling/timing

Screenshot:

What the email looks like for the youth

### 🏕️ (Optional) Future Work / Notes

- Email subject/body (I just made it up for now, should be changed to what we actually want it to be)

### CODE REVIEW

**Review by rymaju — APPROVED** [review]
Nice, are we using the built in SES wrapper for nodemailer?

**Comment by jgfino** [comment]
> Nice, are we using the built in SES wrapper for nodemailer?

Not currently, just plain old nodemailer. I didn't realize SES came with nodemailer we could switch to that to maybe avoid an extra package.

File: src/assignment/assignment.service.ts [hunk 1]
```diff
@@ -95,4 +106,70 @@ export class AssignmentService {
...
+  async sendUnsentSurveysToYouth(): Promise<void> {
+    const unsentAssignments = await this.assignmentRepository.find({
+      relations: ['youth', 'reviewer'],
+      where: { sent: false, status: AssignmentStatus.COMPLETED },
+    });
+
+    await Promise.all(
+      unsentAssignments.map(
+        async (assignment) =>
+          assignment.youth.role === YouthRoles.TREATMENT && this.sendToYouth(assignment),
```
Comment (onb6) [inline src/assignment/assignment.service.ts:171]: love the abstraction into helpers here, really easy to understand

File: src/util/email/amazon-ses.wrapper.ts [hunk 2]
```diff
@@ -21,30 +28,39 @@ export class AmazonSESWrapper {
...
-            Charset: 'UTF-8',
-            Data: bodyHtml,
-          },
-        },
-        Subject: {
-          Charset: 'UTF-8',
-          Data: subject,
-        },
-      },
+      RawMessage: { Data: messageData },
```
Comment (onb6) [inline src/util/email/amazon-ses.wrapper.ts:61]: Just wondering, did amazon SES not have attachment functionality? This looks good, just want to be sure we aren't using more dependencies than necessary

**Comment by onb6** [comment]
> > Nice, are we using the built in SES wrapper for nodemailer?
> 
> Not currently, just plain old nodemailer. I didn't realize SES came with nodemailer we could switch to that to maybe avoid an extra package.

I see now you already said the same thing I left a comment on lol! Yeah I would look into that, but all in all this looks great!

**Review by onb6 — APPROVED** [review]
Really great work on this!

**Pull request merged**